### DISCUSSION TITLE
How do you use Firebird inside container?

### DISCUSSION THREAD

**Discussion by gsbelarus** [FirebirdSQL/firebird#8127] [General]
Hi,

I'm completely stuck at the very easy task. Running nextjs app inside a container, The application 
connects to the Firebird using node-firebird-driver.

Are there any recipes? Because of now, I can not move past driver compilation step.
It requires glibc 2.32 and OS has installed 2.35 version.

**Comment by gsbelarus** [comment]
There would be no problem if Firebird had supported HTTP. Using binary library is a great pain in the world of multiple linux distributives and versions. Spending last year in the mongodb world, I guarantee no one in a sane mind would spend his precious time struggling with binaries.

**Comment by aafemt** [comment]
Andrej Kirejeŭ wrote 23.05.2024 22:51:
> There would be no problem if Firebird had supported HTTP.

   Firebird supports TCP. It is enough. It can be anywhere where your Node.JS 
container can reach using TCP attachment.

-- 
   WBR, SD.

**Comment by gsbelarus** [comment]
It can't be anywhere because node-firebird-driver needs binaries. So, I can't separate the web server from the firebird client library. Windows is much easier in that case. We just distributed fbclient.dll and a couple of files along the executable and that is it. But in the world of docker and multiple linux distros, it just doesn't work. And base layer for building node based images is now alpine linux. Is it possible to somehow use client library on alpine? I just don't want to spend half of the life just solving stupid puzzles.

If TCP is just enough then guess why Postgresql has an HTTP access since 2015and MySQL since version 6?

**Reply by aafemt** [reply]
Perhaps because someone paid them for it...?

**Comment by asfernandes** [comment]
> Hi,
> 
> I'm completely stuck at the very easy task. Running nextjs app inside a container, The application connects to the Firebird using node-firebird-driver.
> 
> Are there any recipes? Because of now, I can not move past driver compilation step. It requires glibc 2.32 and OS has installed 2.35 version.

This topic is in wrong place, totally incomplete, imprecise and probably wrong.

I can't manage to help if you not help others understand you, or if you want to spend only 5 seconds to explain your problems.

**Comment by asfernandes** [comment]
> It can't be anywhere because node-firebird-driver needs binaries.

There is a reason project is split in node-firebird-driver and node-firebird-driver-native, so the version that don't need to depent on "native" libraries can be done. So, one need to do it. Or ask tha AI to create it haha.


> So, I can't separate the web server from the firebird client library.

And what?


> Windows is much easier in that case.

If you understand Windows only, go for it.


> We just distributed fbclient.dll and a couple of files along the executable and that is it.

Nice, then use it.


> But in the world of docker and multiple linux distros, it just doesn't work.

You can start study Linux and Docker.


> And base layer for building node based images is now alpine linux.

So don't use it.


> Is it possible to somehow use client library on alpine?

Or use it (alpine-based node) with a Firebird built in alpine.


> I just don't want to spend half of the life just solving stupid puzzles.
> 

In your opinion, you mean!

Then use what you understand.


> If TCP is just enough then guess why Postgresql has an HTTP access since 2015and MySQL since version 6?

I guess because someone wanted to created it.

**Comment by asfernandes** [comment]
Hint: https://github.com/FirebirdSQL/firebird/blob/master/.github/workflows/main.yml#L123
